src/gov-action-contracts/governance/SetProposalThresholdsAndConstitutionHashAction.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import "../address-registries/L2AddressRegistry.sol";
+import "./SetProposalThresholdLib.sol";
+
+// @notice Governance action for AIP 1.2 https://forum.arbitrum.foundation/t/proposal-aip-1-2-foundation-and-dao-governance/13362/22
+contract SetProposalThresholdsAndConstitutionHashAction {
+    IL2AddressRegistry public immutable l2GovAddressRegistry;
+
+    bytes32 public constant newConstitutionHash = bytes32("0xTODO"); // TODO: update value
+    uint256 public constant newProposalThreshold = 1_000_000 ether;
+
+    constructor(IL2AddressRegistry _l2GovAddressRegistry) {
+        l2GovAddressRegistry = _l2GovAddressRegistry;
+    }
+
+    function perform() external {
+        IL2ArbitrumGoverner coreGov = l2GovAddressRegistry.coreGov();
+        SetProposalThresholdLib.setProposalThreshold(coreGov, newProposalThreshold);
+        require(
+            coreGov.proposalThreshold() == newProposalThreshold,
+            "SetProposalThresholdsAndConstitutionHashAction: core governer proposal threshold not set"
+        );
```

Comment:
whats the point of this check?
and if we're doing it, do we also want a check on the previous threshold before executing the setThreshold?

Reply:
just a sanity check eh

Reply:
Yeh, it doesnt do anything really. It's just extra assurance that `setProposalThreshold` behaved the way you expected - we do a `relay` call in there, so this extra assurance is kinda nice. Not strictly necessary, but also shouldnt be harmful.